### PR TITLE
Assorted fixes

### LINKED ISSUE #1: Missing `dyne` unit
## Description

The `defaults.txt` file references a [`dyne` unit](https://github.com/mattwthompson/openff-units/blob/0d9d906b325fea0411fb4f4a0aea06cfc8ca813e/openff/units/data/defaults.txt#L121) but this is not defined anywhere.

### LINKED ISSUE #2: Cleanup files
## Description

If this package goes into production any unused files should probably be removed and the READMEs fleshed out. In particular:

* [`look_and_say`](https://github.com/mattwthompson/openff-units/blob/main/openff/units/data/look_and_say.dat) should be removed
* The data [`README`](https://github.com/mattwthompson/openff-units/blob/main/openff/units/data/README.md) file should be updated to briefly state the provenance of the constants and defaults file (i.e. pint representation of CODATA XXXX).

The main `README` should be updated to probably just say something along the lines of:

> This package provides a common unit registry for all OpenFF packages to use in order to ensure consistent unit definitions across the software ecosystem`. 
> 
> The unit definitions are currently sourced from the NIST XXXX CODATA, but may be updated in future versions as new CODATA updates are made.
> 
> While this repository is based on `pint`, the main `Unit`, `Quantity` and `Measurement` classes have been subclassed in order to provide non-dynamic, more readily serialisable representations.
>
> ## Getting Started
>
> SHORT EXAMPLE HERE

### LINKED ISSUE #3: Vendor `simtk` unit to string code
## Description

Currently the `utils` module imports the [`unit_to_string`](https://github.com/mattwthompson/openff-units/blob/0d9d906b325fea0411fb4f4a0aea06cfc8ca813e/openff/units/utils.py#L26) method from the [OpenFF toolkit](https://github.com/openforcefield/openff-toolkit/blob/24953a407c853411bee9584d29fa0fb953e59151/openff/toolkit/utils/utils.py#L208). It would seem however best to just vendor it here instead given that the TK would likely eventually depend on this package.

### PR DESCRIPTION
**Pull request by mattwthompson**
Fixes #1 
Fixes #2
Fixes #3

- [x] Remove un-used files
- [x] Update main README with code snippet
- [x] Update data README to denote CODATA version
- [x] Update for changes to `openff-utilities`
- [x] `utils.py` -> `utilities.py`
- [x] Add dyne to get erg working
- [x] Vendor `unit_to_string`

### CODE REVIEW

File: openff/units/data/README.md [hunk 1]
```diff
@@ -1,21 +1,3 @@
...
-## Including package data
-
-Modify your package's `setup.py` file and the `setup()` command. Include the 
-[`package_data`](http://setuptools.readthedocs.io/en/latest/setuptools.html#basic-use) keyword and point it at the 
-correct files.
-
-## Manifest
-
-* `look_and_say.dat`: first entries of the "Look and Say" integer series, sequence [A005150](https://oeis.org/A005150)
+The unit definitions are currently sourced from the NIST 2018 [CODATA](https://physics.nist.gov/cuu/Constants/), but may be updated in future versions as new CODATA updates are made.
```
Comment (mattwthompson) [inline openff/units/data/README.md:3]: I would like confirmation this is accurate.   I copied the file from Evaluator

**Comment by mattwthompson** [comment]
I'm only round-tripping through pickle `Quantity` objects, not `Unit` or `Measurement`, causing the coverage to be not-so-great: https://codecov.io/gh/mattwthompson/openff-units/src/533ca5f4bcec72e7ca852c0bedfc4d6cf7f8e9bd/openff/units/units.py

File: openff/units/data/README.md (hunk 1 above)
Comment (SimonBoothroyd) [reply openff/units/data/README.md:3]: Yes I believe it is.

**Review by SimonBoothroyd — APPROVED** [review]
Thanks for making these changes @mattwthompson - these all LGTM!

**Pull request merged**